pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1839,9 +1840,17 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(ctx context.Context, clusterName
 			if err != nil {
 				return fmt.Errorf("failed to get floating IP for loadbalancer VIP port %s: %v", portID, err)
 			}
+
+			// Delete the floating IP only if it was created dynamically by the controller manager.
 			if fip != nil {
-				if err := floatingips.Delete(lbaas.network, fip.ID).ExtractErr(); err != nil {
-					return fmt.Errorf("failed to delete floating IP %s for loadbalancer VIP port %s: %v", fip.FloatingIP, portID, err)
+				matched, err := regexp.Match("Floating IP for Kubernetes external service", []byte(fip.Description))
+				if err != nil {
+					return err
+				}
+				if matched {
+					if err := floatingips.Delete(lbaas.network, fip.ID).ExtractErr(); err != nil {
+						return fmt.Errorf("failed to delete floating IP %s for loadbalancer VIP port %s: %v", fip.FloatingIP, portID, err)
+					}
 				}
```

Comment:
should we add some log in !matched flag ?

Reply:
We could, but I didn't see it's very useful, it's obvious that only the floating IP whose description not satisfies the regex will not be touched. The condition answer is yes or no, the log is useful only if there are multiple conditions.

Reply:
Hi @jichenjc, are you ok if we merge this PR?